test/e2e/apimachinery/garbage_collector.go

```diff
@@ -445,17 +446,13 @@ var _ = SIGDescribe("Garbage collector", func() {
 			framework.Failf("%v", err)
 		}
 		By("wait for 30 seconds to see if the garbage collector mistakenly deletes the pods")
-		if err := wait.Poll(5*time.Second, 30*time.Second, func() (bool, error) {
-			pods, err := podClient.List(metav1.ListOptions{})
-			if err != nil {
-				return false, fmt.Errorf("Failed to list pods: %v", err)
-			}
-			if e, a := int(*(rc.Spec.Replicas)), len(pods.Items); e != a {
-				return false, fmt.Errorf("expect %d pods, got %d pods", e, a)
-			}
-			return false, nil
-		}); err != nil && err != wait.ErrWaitTimeout {
-			framework.Failf("%v", err)
+		time.Sleep(30 * time.Second)
```

Comment:
nit: polling helps us fail earlier in case of error. but in reality we expect these tests to pass consistently, so there is no difference